lib/training_data.py

```diff
@@ -13,25 +20,63 @@ def __init__(self, random_transform_args, coverage, scale=5, zoom=1): #TODO thos
         self.zoom = zoom
 
     def minibatchAB(self, images, batchsize):
-        batch = BackgroundGenerator(self.minibatch(images, batchsize), 1)
+    
+        def execute_yaw_sorter ( input_dir, output_path ):
+            yaw_sorter_process = subprocess.Popen(
+                        sys.executable + " " +                
+                        os.path.join ( os.path.dirname(__file__), 'yaw_sorter.py' ) + ' "' + input_dir + '" "' + output_path + '"'
+                        , env=os.environ)                        
+            yaw_sorter_process.wait()
+        
+        images_dir = os.path.dirname (images[0])
+        images_parent_dir = os.path.join ( images_dir, os.pardir )
+        images_parent_dir_alignments_yaw_json = os.path.join ( images_parent_dir, "alignments_yaw.json" )
+        

```

Comment:
There is a bug where if there is no `alignments_yaw.json` in the `images_parent_dir`, then the `while` block will throw a `FileNotFoundError` when trying to run `execute_yaw_sorter` on line 40. 
The `alignments_yaw.json` can be empty with at least `{}` as the only contents.
I think this is what the people saying that `alignments_yaw.json` being required now were referring to.

An easy fix is:
```python
if not os.path.isfile(images_parent_dir_alignments_yaw_json):
    with open(images_parent_dir_alignments_yaw_json, 'w') as f:
        f.write('{}')
```

Reply:
`execute_yaw_sorter (images_dir, images_parent_dir_alignments_yaw_json )`

will create file, so 
`not os.path.isfile(images_parent_dir_alignments_yaw_json)` will not occur